### PR TITLE
Made some changes to households.py

### PR DESCRIPTION
**Pull request by jtregde**
I made the changes to `FOCs` that @LeiliPR recommended. I also renamed what we had been calling `nu` to `upsilon` to be in keeping with the documentation of the elliptical estimation file.

### CODE REVIEW

File: scripts/households.py [hunk 1]
```diff
@@ -1,66 +1,65 @@
...
+    # wpath = length p, b_sp1 = length p-1
 
-    foc_errors = np.append(foc_errors_b, foc_errors_n)
+    b_s = np.append(b_init, b_sp1)
+    b_sp1 = np.append(b_sp1, 0.0)
 
-    return foc_errors
+    c = (1 + rpath) * b_s + wpath * n + BQpath - b_sp1
+    MU_c = mu_cons(c, sigma)
+    errors = (MU_c[:-1] - beta * (1 + rpath[1:]) *
```
Comment (jdebacker) [inline scripts/households.py]: This is just the FOC for the choice of savings.  Why deleting the FOC for the choice of labor supply?

File: scripts/households.py [hunk 2]
```diff
@@ -1,66 +1,65 @@
...
+    # rpath = length p, n = length p
+    # p = upsilonmber of periods left in life time
+    # wpath = length p, b_sp1 = length p-1
 
-    foc_errors = np.append(foc_errors_b, foc_errors_n)
+    b_s = np.append(b_init, b_sp1)
+    b_sp1 = np.append(b_sp1, 0.0)
 
-    return foc_errors
+    c = (1 + rpath) * b_s + wpath * n + BQpath - b_sp1
```
Comment (jdebacker) [inline scripts/households.py]: You have a `get_c` function you could use... I get that it's simple.  But it's often nice to break the code up into those smaller functions (e.g., a function for each equation in the model).  In this way, you can test individual functions to see that they return what you expect given the inputs.  I'll note that this `FOC` function that includes both the labor and savings FOCs can work, but will be more difficult to debug then having the FOC function call two separate function: say `FOC_labor` and `FOC_save` or something.  Each of which you can test independently quite easily to see that they return the expected errors for given inputs...

Comment (jtregde) [reply scripts/households.py]: @jdebacker something like what I just pushed?

Comment (jdebacker) [reply scripts/households.py]: Ok - yes, you've separated them nicely.  But you'll still need to solve them jointly.  So you'll want to keep a function that looks something like:

```
def FOCs(some args):
    c = get_c(some args)
    n_errors = FOC_labor(c, some args)
    b_errors = FOC_save(c, some args)
    errors = np.append(n_errors, b_errors)  # use whatever order you had before

   return errors
```

**Comment by adibalaram** [comment]
Why is `households.py` being changed? The earlier version had both b and n being solved for simultaneously and the get_c function did what was needed. Someone seems to have changed upsilon back to nu from what I had last. I'm currently making sure that this script has everything that's needed, with the main missing component being that there's no function call to obtain BQ. I'm pretty certain we don't need to add anything else to this script (at least for SS to work).

**Comment by jdebacker** [comment]
@adibalaram Thanks for reviewing this.  I can see that there need to be some additions here before things can be merged.  I'll wait for those and then for you to sign off that this PR is ready to merge.

**Comment by jtregde** [comment]
Just glanced at the new PR @adibalaram just made. I think that one renders this one unnecessary now. If we still want to split the `FOCs` function into separate `FOC_labor` and `FOC_save` functions as @jdebacker suggested, I think we can do it from that one.

**Comment by jdebacker** [comment]
@jtregde I like what you've done here. When this is done (and @adibalaram gives me a thumbs up), I'll merge.  He can then resolve any conflicts on his branch to continue to make his PR relevant.

**Comment by adibalaram** [comment]
I'm confused by why we've shifted to r_path instead of just r - is there a difference between the two? Also, why is it being indexed from 1 for the computation of b_s? There's just r being used in the computation of n_s. I would think that `household.py` doesn't need to be modified specifically for TPI given the way we did it in class. Sorry for all the confusion! I'm just trying to make sure our variable names are consistent.

**Comment by jtregde** [comment]
So the way I was thinking of it, is that households need to know the whole path of r in order to make savings decisions (like in equation 8.11). Is this right? Or is it just that they need each period's interest rate to make each period's decision?

**Comment by adibalaram** [comment]
I think it makes more sense to just have `r` in the households.py file. That's what we did in the 3-period model too. We can ensure that what is passed to this file from TPI.py is `r_path` indexed appropriately. Does that make sense?

The downside with using some element of `r_path` is that solving the steady state then becomes tricky. All we're doing in SS.py is choosing an r and solving the HH and firm problems repeatedly until we converge. This gets hard (at least for me) if we're feeding in some path for r.

**Comment by jdebacker** [comment]
Yes, I think the way we wrote the household functions for the 3-period model was a way in which one could pass `r` as a scalar or vector.  So if you keep that structure, it's just a matter of being consistent with the name.  `r` is shorter and it could be a scalar or vector, so maybe that makes sense here.  

I typically only add some suffix to the variable name (`_mat`, `_vec`, `_path`) if there's a chance it could be confused with something with a similar name, but a different shape.

**Comment by adibalaram** [comment]
Thanks a lot @jtregde. @jdebacker This is ready to go!

**Comment by jtregde** [comment]
Thanks for checking that for me and catching those mistakes @adibalaram!

**Comment by jdebacker** [comment]
@jtregde et al.  I made a PR to this branch and (1) cleaned up some HH functions (2) added tests for all household function and (3) added a spreadsheet with the calculations I used to find the expected values for the tests.

Note that you may need to continue to update `households.py`, but you want to continue to pass these tests (note that you may need to adjust arguments to the test functions).

**Pull request merged**